src/test/scala/edu/ie3/simona/service/primary/PrimaryServiceWorkerSqlIT.scala

```diff
@@ -0,0 +1,219 @@
+import edu.ie3.simona.ontology.messages.SchedulerMessage.{
+  CompletionMessage,
+  ScheduleTriggerMessage,
+  TriggerWithIdMessage
+}
+import edu.ie3.simona.ontology.messages.services.ServiceMessage.RegistrationResponseMessage.RegistrationSuccessfulMessage
+import edu.ie3.simona.ontology.messages.services.ServiceMessage.WorkerRegistrationMessage
+import edu.ie3.simona.ontology.trigger.Trigger.{
+  ActivityStartTrigger,
+  InitializeServiceTrigger
+}
+import edu.ie3.simona.service.primary.PrimaryServiceWorker.{
+  ProvidePrimaryDataMessage,
+  SqlInitPrimaryServiceStateData
+}
+import edu.ie3.simona.test.common.AgentSpec
+import edu.ie3.util.TimeUtil
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.testcontainers.utility.MountableFile
+
+import java.nio.file.Paths
+import java.util.UUID
+import scala.language.postfixOps
+import scala.reflect.ClassTag
+
+class PrimaryServiceWorkerSqlIT
+    extends AgentSpec(
+      ActorSystem(
+        "PrimaryServiceWorkerSqlIT",
+        ConfigFactory
+          .parseString("""
+                     |akka.loglevel="OFF"
+          """.stripMargin)
+      )
+    )
+    with ForAllTestContainer
+    with BeforeAndAfterAll
+    with TableDrivenPropertyChecks {
+
+  override val container: PostgreSQLContainer = PostgreSQLContainer(
+    "postgres:11.14"
+  )
+
+  private val simulationStart =
+    TimeUtil.withDefaults.toZonedDateTime("2020-01-01 00:00:00")
+
+  private val schemaName = "public"
+
+  private val uuidP = UUID.fromString("9185b8c1-86ba-4a16-8dea-5ac898e8caa5")
+  private val uuidPhq = UUID.fromString("46be1e57-e4ed-4ef7-95f1-b2b321cb2047")
+
+  private val tableNameP = s"its_p_$uuidP"
+  private val tableNamePhq = s"its_pqh_$uuidPhq"
+
+  override protected def beforeAll(): Unit = {
+    val url = getClass.getResource("timeseries/")
+    url shouldNot be(null)
+    val path = Paths.get(url.toURI)
+
+    // Copy sql import scripts into docker
+    val sqlImportFile = MountableFile.forHostPath(path)
+    container.copyFileToContainer(sqlImportFile, "/home/")
+
+    Iterable(s"$tableNameP.sql", s"$tableNamePhq.sql")
+      .foreach { file =>
+        val res = container.execInContainer("psql", "-Utest", "-f/home/" + file)
+        res.getStderr shouldBe empty
+      }
+  }
+
+  override protected def afterAll(): Unit = {
+    container.stop()
+    container.close()
+  }
+
+  // asInstanceOf throws ClassCastException if cast fails, thus this is safe here
+  @SuppressWarnings(Array("AsInstanceOf"))
+  private def getServiceActor[T <: Value](
+      scheduler: ActorRef
+  )(implicit tag: ClassTag[T]): PrimaryServiceWorker[T] = {
+    new PrimaryServiceWorker[T](
+      scheduler,
+      tag.runtimeClass.asInstanceOf[Class[T]],
+      simulationStart
+    )
+  }
+
+  "A primary service actor with SQL source" should {
+    "initialize and send out data when activated" in {
+
+      val cases = Table(
+        (
+          "getService",
```

Comment:
Well... Passing around objects and naming them like methods... I don't really trust this thing... Maybe you use partial functions instead!?